### ISSUE TITLE
Generate PROTOCOL_VERSION

### ISSUE THREAD

**Issue by Roman-** [microavia/messgen#50]
For each generator, add PROTOCOL_VERSION calculator. This will make it easy to compare the protocol versions between the two peers.

# Workaround
Here's a workaround we use now in our project:
- the .yaml files are in the separate repo. Backend and frontend use this repo as submodule
- upon build, hash of last commit in the submodule is considered a protocol version
- on the first handshake request, Backend sends its protocol version to Frontend
- Frontend checks the hash of the last commit of its submodule and compares the two

# Workaround limitations
- it is limited to one messgen protocol per submodule. If multiple protocols share a repo, they also share the last commit hash
- unrelated changes in submodule (e.g. change README) also change the protocol version
- when you make changes in your .yaml files without commiting them, the protocol is considered unchanged

# Suggested approach
For each protocol, its version must be calculated as md5 of all .yaml files in protocol directory. Changing any .yaml file will change protocol md5. Using the truncated md5 (e.g. first 10 chars only) should also be fine.

For cpp generator, here's an example of what it should genarate in `messages.h`:
```
struct ProtoInfo {
    static constexpr uint8_t ID = 1;
    static constexpr uint32_t MAX_MESSAGE_SIZE = 24;
    static constexpr const char* VERSION = "i2e8nb0a";
};
```

**Comment by meded90** [comment]
https://github.com/microavia/messgen/pull/51
resolve task

**Comment by Roman-** [comment]
Closed by #51 and #52.
Thanks @meded90 !
